### ISSUE TITLE
Triples are not persisted correctly when the graphs are first created

### ISSUE THREAD

**Issue by pinnaculum** [RDFLib/rdflib#2461]
In [galacteek](https://gitlab.com/galacteek/galacteek) i use rdflib, there's a big conjunctive graph with subgraphs. The hierarchy of the graphs can be [seen here](https://gitlab.com/galacteek/galacteek/-/raw/master/galacteek/services/ld/pronto/config.yaml), see the *urn:ipg:i* top graph. [Service code here](https://gitlab.com/galacteek/galacteek/-/blob/master/galacteek/services/ld/pronto/__service__.py). RDF datasets are downloaded and added/merged in some of those subgraphs. The store plugin is Sleepycat.

It all works out pretty well, except i've noticed an issue on the first run, when the graphs don't exist yet. In that case, the data from the datasets is not correctly commited. When the app is restarted, the datasets are downloaded again, merged, and this time are commited properly, and all the triples are there. I always *.commit()* on the graph after each merge.

What could cause this problem to only manifest itself on the first run (with no graph db yet) and not on following runs ? I've tried different versions of rdflib, and different store plugins, same problem.

Thank you, and sorry in advance if i didn't provide enough infos. [Original issue on gitlab](https://gitlab.com/galacteek/galacteek/-/issues/15).

**Comment by aucampia** [comment]
Its not that easy to tell, but from the codebase it seems like you are using a rather old version of RDFLib. Can you try with RDFLib 7.0.0?

**Comment by pinnaculum** [comment]
@aucampia Thank you. Indeed, the rdflib that i use is a customized version of rdflib 5.0, which is now pretty old. I've changed it to be able to load schemas from IPFS. What i'll try to do, is write a small program that reproduces what galacteek does with the conjunctive graphs, and try it with rdflib 7.0. Migrating ro rdflib 7.0 is a bit difficult because i've also patched rdflib-jsonld ^_^ which seems to have been merged in rdflib now ?

I don't know if this "layout" of graphs is possible in theory with rdflib ? i haven't seen any example do subconjunctive graphs, but it does work after the 2nd run, which is really confusing :)
